src/server_manager/web_app/ui_components/outline-per-key-data-limit-dialog.ts

```diff
@@ -269,100 +262,82 @@ export class OutlinePerKeyDataLimitDialog extends LitElement {
       }
       const inputLimit = displayDataAmountToBytes(this.inputDataLimit());
-      if (inputLimit !== this._keyDataLimitBytes) {
+      if (inputLimit !== this._activeDataLimitBytes) {
         return Change.SET;
       }
       return Change.UNCHANGED;
     }
-    if (this._keyDataLimitBytes !== undefined) {
+    if (this._activeDataLimitBytes !== undefined) {
       // The user must have unchecked the checkbox.
       return Change.REMOVED;
     }
     return Change.UNCHANGED;
   }
 
-  /**
-   * Returns true if the data limit was changed by the user.
-   */
-  public dataLimitChanged() {
-    return this._dataLimitChange() !== Change.UNCHANGED;
-  }
-
   /**
    * The current data limit as input by the user, but not necessarily as saved.
    */
   public inputDataLimit(): DisplayDataAmount {
-    return this._showMenu ? {unit: this._dataLimitType(), value: this._dataLimitValue()} : null;
-  }
-
-  /**
-   * The ID of the key being worked on.  Useful for making API requests for the given key.
-   */
-  public keyId(): string {
-    return this._keyId;
+    return this._showDataLimit ? {unit: this._dataLimitUnit(), value: this._dataLimitValue()} :
+                                 null;
   }
 
   /**
-   * The ID of the server the current key belongs to.
-   */
-  public serverId(): string {
-    return this._serverId;
-  }
-
-  /**
-   * Opens the dialog to display data limit information about the given key
+   * Opens the dialog to display data limit information about the given key.
    *
    * @param keyName - The displayed name of the access key.
-   * @param keyId - The unique ID of the access key.
    * @param keyDataLimit - The display data limit of the access key, or undefined.
-   * @param serverId - The ID of the server the access key is a part of.
    * @param serverDefaultLimit - The default data limit for the server, or undefined if there is
-   *     none
+   *     none.
+   * @param language - The app's current language
+   * @param onDataLimitSet - Callback for when a data limit is imposed.  Must return whether or not
+   *    the data limit was set successfully.  Must not throw or change the dialog's state.
+   * @param onDataLimitRemoved - Callback for when a data limit is removed.  Must return whether or
+   *    not the data limit was removed successfully.  Must not throw or change the dialog's state.
    */
   public open(
-      keyName: string, keyId: string, keyLimitBytes: number, serverId: string,
-      defaultLimitBytes: number, language: string) {
+      keyName: string, keyLimitBytes: number, onDataLimitSet: OnSetDataLimitHandler,
+      onDataLimitRemoved: OnRemoveDataLimitHandler) {
     this._keyName = keyName;
-    this._keyId = keyId;
-    this._keyDataLimitBytes = keyLimitBytes;
-    this._serverId = serverId;
-    this._serverDefaultLimitBytes = defaultLimitBytes;
-    this._language = language;
-    this.setInitialMenuState();
+    this._activeDataLimitBytes = keyLimitBytes;
```

Comment:
Let's rename this to `initialDataLimitBytes`. It makes it clearer it was the initial value. "Active" is a higher level concept.

Reply:
Done in 3a84a661f51d59193d732ae58743c5e6c7075746